src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -257,11 +260,22 @@ bool java_bytecode_languaget::typecheck(
     get_message_handler());
 
   // now typecheck all
-  if(java_bytecode_typecheck(
-       symbol_table, get_message_handler(), string_refinement_enabled))
-    return true;
+  auto res = java_bytecode_typecheck(
```

Comment:
Formatting - please use the linter. Also type is not implied by rhs, please use actual type

Reply:
If the result is bad (you'll have to check `java_bytecode_typecheck` to work out whether that is `true` or `false`), you probably shouldn't bother with generating the new symbols as the program is about to terminate!